src/ui/src/core_components/other/CoreJsonViewer.vue

```diff
@@ -0,0 +1,130 @@
+<template>
+	<div>
+		<p v-if="error" class="error">JSON Viewer error: {{ error }}</p>
+		<WdsJsonViewer
+			v-else-if="data"
+			ref="rootInstance"
+			:data="data"
+			:paths-open="pathsOpen"
+			@toggle="onToggle"
+		/>
+		<WdsJsonViewerChildrenCounter v-else :data="{}" />
+	</div>
+</template>
+
+<script lang="ts">
+import { cssClasses, separatorColor } from "../../renderer/sharedStyleFields";
+import { JsonViewerTogglePayload } from "../../wds/WdsJsonViewer.utils";
+import { FieldCategory, FieldType } from "../../writerTypes";
+
+const description = "A standalone element to view a JSON data.";
+
+const initialData = {
+	name: "JSON Viewer",
+	description: "A JSON tree viewer were you can expand key",
+	sample: {
+		description: "This sample is opened by default",
+		bool: true,
+		null: null,
+		list: [1, "two", { key: 3 }],
+	},
+	sampleClosed: {
+		description: "This sample is not opened by default",
+	},
+	createdAt: new Date(),
+};
+
+const toggleHandlerStub = `
+def handle_json_tree_toggle(state, payload):
+
+	# string array representing the keys of the path
+	path = payload.get("path")
+
+
+	# boolean if the path was open (True) or closed (False)
+	open = payload.get("open")
+
+	# ...`;
+
+export default {
+	writer: {
+		name: "JSON Viewer",
+		description,
+		category: "Other",
+
+		events: {
+			"wf-toggle": {
+				desc: "Capture JSON tree element open.",
+				stub: toggleHandlerStub.trim(),
+			},
+		},
+
+		fields: {
+			data: {
+				name: "Data",
+				init: JSON.stringify(initialData),
+				type: FieldType.Text,
+			},
+			pathsOpen: {
+				name: "Paths open",
+				init: JSON.stringify({ "sample.list.2": true }),
+				type: FieldType.KeyValue,
+			},
+			collapseAngleColor: {
+				name: "Collapse angle",
+				type: FieldType.Color,
+				category: FieldCategory.Style,
+				applyStyleVariable: true,
+			},
+			jsonViewerValueColor: {
+				name: "JSON value",
+				type: FieldType.Color,
+				category: FieldCategory.Style,
+				applyStyleVariable: true,
+			},
+			jsonViewerIndentationSpacing: {
+				name: "JSON indentation",
+				type: FieldType.Width,
+				category: FieldCategory.Style,
+				applyStyleVariable: true,
+			},
+			separatorColor,
+			cssClasses,
+		},
+		previewField: "data",
+	},
+};
+</script>
+
+<script setup lang="ts">
+import { ComponentPublicInstance, computed, inject, ref } from "vue";
+import injectionKeys from "../../injectionKeys";
+import WdsJsonViewer from "../../wds/WdsJsonViewer.vue";
```

Comment:
**Important: Don't use WDS for this.** To share Vue components across Writer Framework components use "base". So that'll require some moving/renaming.

WDS is the Writer Design System and should match 1-1 what's in the design system (which we don't control, the designers do).

Reply:
The thing is, they are components I use only to build `CoreJsonViewer`. `src/ui/src/core_components/base` contains component that are exposed by `templateMap.ts`.

As I moved `CoreJsonViewer` as `content`, I think it would makes sens to move those internal components to `src/ui/src/core_components/content/components`. Thus, the `components` folder expose internal components for `content` components.

WDYT ?

Reply:
The ones in `base` aren't exposed by `templateMap.ts`. They're in fact "base" Vue components and not for "direct consumption".

Reply:
Ha, I see. Got it :+1: